src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -246,15 +246,18 @@ def arguments_loader():
     def handler(args):  # pylint: disable=too-many-branches,too-many-statements
         from msrestazure.azure_operation import AzureOperationPoller
 
-        ordered_arguments = args.pop('ordered_arguments') if 'ordered_arguments' in args else []
+        ordered_arguments = args.pop('ordered_arguments', [])
+        for item in ['properties_to_add', 'properties_to_set', 'properties_to_remove']:
+            if args[item]:
+                raise CLIError("Unexpected '{}' was not empty.".format(item))
+            del args[item]
 
         try:
             client = factory() if factory else None
         except TypeError:
             client = factory(None) if factory else None
 
-        getterargs = {key: val for key, val in args.items()
-                      if key in get_arguments_loader()}
+        getterargs = {key: val for key, val in args.items() if key in get_arguments_loader()}
```

Comment:
We are only calling it once in the handler. This method does get the "argument_loader" in the sense that "get_foo" would get the foo.  It returns the CLI Arguments for the "get" side of the update.  There's a corresponding "set_arguments_loader" that returns the CLI Arguments for the "set" side of the update.  That's also only called once in the handler.